### PR TITLE
#149646431 Add appropriate error responses

### PR DESCRIPTION
**Pull request by danielwangai**
#### What does this PR do?

Add descriptive error messages

#### Description of Task to be completed?

Add error messages working on the following endpoints

`PUT api/v1/paths/<id>` - if id is not provided the message below is returned.

```
{
    "error": "Path id must be provided"
}
```

`DELETE api/v1/paths/<id>` - if id is not provided the message below is returned.

```
{
    "error": "Path id must be provided"
}
```

If an unknown parameter is passed during a POST or PUT request, the following error message is added.

```
{
    "error": "<Wrong-Param-ID> is not a required parameter."
}
```

#### How should this be manually tested?

`pytest`

#### What are the relevant pivotal tracker stories?

#149646431

### CODE REVIEW

**Comment by danielwangai** [comment]
@njirap Commit squashed and rebased to current `develop`.

File: api/views/paths.py [hunk 1]
```diff
@@ -8,12 +8,25 @@ class PathResource(Resource):
...
+
+    def __init__(self):
+        self.path_params = ["id", "name", "description"]
+        
     def post(self):
         """
         To create a learning path
         """
         if request.get_json():
             result = request.get_json()
```
Comment (njirap) [inline api/views/paths.py]: maybe a better var name

Comment (njirap) [inline api/views/paths.py]:
```
payload = request.get_json()
if payload:
```

File: api/views/paths.py [hunk 2]
```diff
@@ -8,12 +8,25 @@ class PathResource(Resource):
...
+            for param in result.keys():
+                if param not in self.path_params:
+                    return {"error": "<{}> is not a required parameter.".format(
```
Comment (njirap) [inline api/views/paths.py]: this appears rather hash

Comment (njirap) [inline api/views/paths.py]: just pick everything you want and ignore the rest if any

File: api/views/paths.py [hunk 3]
```diff
@@ -8,12 +8,25 @@ class PathResource(Resource):
...
+                        param
+                    )}, 400
+            if not result["name"]:
+                return {"error": "Name is required."}, 400
```
Comment (njirap) [inline api/views/paths.py]:
```
name = result['name']
if not name: return {"error": "Name is required."}, 400

File: api/views/paths.py [hunk 4]
```diff
@@ -8,12 +8,25 @@ class PathResource(Resource):
...
+            if not result["description"]:
+                return {"error": "Description is required."}, 400
```
Comment (njirap) [inline api/views/paths.py]:
```
if not (name or description):
    return noth name and description must not be blank
```

File: api/views/paths.py [hunk 5]
```diff
@@ -8,12 +8,25 @@ class PathResource(Resource):
...
             if Path.query.filter_by(name=result["name"]).first():
                 # prevent instances of duplicate path names
                 return {"error": "A Path with same name exists."}, 409
```
Comment (njirap) [inline api/views/paths.py]: return the name they passed

Comment (njirap) [inline api/views/paths.py]: make responses uniform in data types and data fields

File: api/views/paths.py [hunk 6]
```diff
@@ -60,12 +73,21 @@ def get(self, id=None):
...
                 })
 
-    def put(self, id):
+    def put(self, id=None):
         """
         To update a path.
         """
         if request.get_json():
             result = request.get_json()
+            if not id:
```
Comment (njirap) [inline api/views/paths.py]: change var name because `id` is a built-in function in python

File: test/api/views/test_paths.py [hunk 7]
```diff
@@ -41,6 +41,50 @@ def test_create_path(self):
         self.assertEqual(len(new_all_paths), len(old_all_paths) + 1)
         self.assertEqual(response.status_code, 201)
 
+    def test_create_path_without_name(self):
```
Comment (njirap) [inline test/api/views/test_paths.py]: test names could be better to imply or suggest the expected behaviour eg `tes_path_name_required`

File: api/views/paths.py (hunk 6 above)
Comment (danielwangai) [inline api/views/paths.py]: Changed the parameter from `id` to `path_id`. This meant that I also changed the id argument in the definition of the endpoint url from `/api/v1/paths/string<id>` to `/api/v1/paths/string<path_id>`

File: api/views/paths.py (hunk 5 above)
Comment (danielwangai) [inline api/views/paths.py]: Changes made. A successful fetch returns a list of paths (if `path_id` is not passed and paths exist) and an empty list if there are no existent paths.

File: api/views/paths.py (hunk 1 above)
Comment (danielwangai) [inline api/views/paths.py]: Implemented on both `PUT` and `POST`.

**Comment by wanjiku-carol** [comment]
@danielwangai 
`api/views/paths.py` line 20 the error message may need correction from this:
> return {"error": "Parameters name and description MUST required."}, 400

**Comment by LarryWachira** [comment]
LGTM

File: test/api/views/test_paths.py [hunk 8]
```diff
@@ -120,6 +166,22 @@ def test_fetch_path_with_invalid_id(self):
...
+        """
+        Test that fetch fails when given invalid id.
+        """
+        path = Path.query.all()[0]
+        response = self.client.put(
+            '/api/v1/paths/',
+            data=json.dumps(self.path2),
+            content_type='application/json'
+        )
+        print(response.__dict__)
```
Comment (njirap) [inline test/api/views/test_paths.py]: what is going on here?

**Pull request merged**